src/preprocessing/preprocess.ts

```diff
@@ -0,0 +1,24 @@
+import { Config } from '../config';
+import { modules as preprocessingModules } from './preprocessing-modules/index';
+
+/**
+ * Runs the preprocessing modules on the given AMP code.
+ *
+ * @param {string} amp AMP code to preprocess
+ * @param {!Config} config Global config
+ * @param {Array<string>=} modules Module whitelist (leave empty to run all)
+ * @return {!Promise<string>} Preprocessed AMP code
+ */
+export async function preprocessAMP(
```

Comment:
do we have a test for this?

Reply:
We do now! Although it will need to be turned into an integration test when the other PRs are merged.